core/multivm_deps/vm_virtual_blocks/src/old_vm/oracles/storage.rs

```diff
@@ -131,6 +137,25 @@ impl<S: WriteStorage, H: HistoryMode> StorageOracle<S, H> {
             .unwrap_or_default()
     }
 
+    fn get_initial_value(&self, storage_key: &StorageKey) -> Option<U256> {
+        self.initial_values.inner().get(storage_key).copied()
+    }
+
+    fn set_initial_value(&mut self, storage_key: &StorageKey, value: U256, timestamp: Timestamp) {
+        match self.get_initial_value(storage_key) {
```

Comment:
`if self.get_initial_value(storage_key).is_none()` is a bit shorter

Also, in the `vm` crate there is already a mechanism to store initial values (only recently added to `boojum-integration`), however, it only adds the initial values only during writes, so when using the `vm` crate you should do similarly to how you do in your PR (set initial values even during reads)